prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,103 @@
+
+## Style Guide
+
+* Alert names MUST be CamelCase, e.g.: `BackendServiceStuck`
+* Alert names SHOULD be prefixed with a component, e.g.: `ZookeeperPersistentVolumeFillingUp`
+  * There may be exceptions for some broadly scoped alerts, e.g.: `TargetDown`
+* Alerts MUST include a `severity` label indicating the alert's urgency.
+  * Valid severities are: `critical` or `warning` — see below for
+    guidelines on writing alerts of each severity.
+* Alerts MUST include `summary` and `description` annotations.
+  * Think of `summary` as the first line of a commit message, or an email
+    subject line.  It should be brief but informative.  The `description` is the
+    longer, more detailed explanation of the alert.
+* Alerts SHOULD include a `namespace` label indicating the source of the alert.
+  * Many alerts will include this by virtue of the fact that their PromQL
+    expressions result in a namespace label.  Others may require a static
+    namespace label.
+* All alerts MUST include a `sop_url` annotation.
+  * Standard Operating Procedure (SOP) style documentation for resolving alerts is required. (SOP Template Placeholder - https://issues.redhat.com/browse/SIGSRE-55)
+
+## Critical Alerts
+
+Critical alerts are for alerting current and impending disaster situations.  These alerts page an SRE.  The situation should warrant waking someone in the middle of the
+night.
+
+Timeline: ~5 minutes.
+
+Reserve critical level alerts only for reporting conditions that may lead to service unavailability.
+Failures of individual components should not trigger critical level alerts in general, unless they would result in that condition.
+Configure critical level alerts so they fire before the situation becomes irrecoverable.
+
+Example critical alert:
+
+```yaml
+- alert: BackendServiceStuck
+  expr: backend_state != 1
+  for: 10m
+  labels:
+    severity: critical
+  annotations:
+    summary: 'The backend service is stuck in a non-ready state'
+    description: 'The backend service {{ $labels.name }} in the {{ $labels.namespace }} namespace, managed by operator {{ $labels.pod }} has been in a non-ready state for 10 minutes'
+    sop_url: 'https://example.com/backend_service_stuck.asciidoc'
+```
+
+This alert fires if a backend service has *not* been ready for the last 10 minutes.
+This is a clear example of a critical issue that represents a threat to the operability of the service, and likely warrants paging someone.
+The alert has a clear summary and description annotations, and it links to a SOP with information on investigating and resolving the issue.
+
+The group of critical alerts should be small, very well-defined, highly documented, polished and with a high bar set for entry.
+
+## Warning Alerts
+
+The vast majority of alerts should use this severity.
+Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the service as a whole.
+
+Timeline: ~60 minutes
```

Comment:
Timeline for action:

Reply:
Similar to above: To what is the "timeline for action" referring?

Reply:
My loose interpretation is that if you don't do something within this timeline, you'll likely have service unavailability.
I guess it's meant to capture the immediacy of the alert.
In these examples it's tying a timeline to a severity. 
Is that a reasonable thing to do?

Reply:
Do we have to mention a timeline for the reaction? I don't believe we should prescribe or mention any number, because these numbers usually vary from team to team.

Reply:
So right now the doc is tying the severity with a specific response timeline.

* Warning => 60mins
* Critical => 5mins

I'm thinking it would be sufficient to leave it ambiguous on the exact timeline (because of reasons discussed above) and just leave this existing wording:

* Warning => Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the service as a whole.
* Critical => Critical alerts are for alerting current and impending disaster situations...

@dastergon @lisa If you're not objectionable I'll make this change

Reply:
sgtm @david-martin